frontend/endgame.py

```diff
@@ -0,0 +1,233 @@
+        """
+        self.profiles[players[0][0]] = {
+            "profile":Image(self.screen, self.profile_image, self.x_percent(556.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen,players[0][1], self.x_percent(556.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status":Image(self.screen, self.pending_image, self.x_percent(664), self.y_percent(503), self.scale_factor, anchor="center")
+            }
+        self.profiles[players[1][0]] = {
+            "profile":Image(self.screen, self.profile_image, self.x_percent(869.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen,players[0][1], self.x_percent(869.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status":Image(self.screen, self.pending_image, self.x_percent(976.5), self.y_percent(503), self.scale_factor, anchor="center")
+            }
+
+    def createThreeProfile(self, players):
+        """Create UI elements for a Three player profile.
+
+        Args:
+            players (list of tuples): A list containing one player tuple in the format (player_id, player_name).
+        """
+        self.profiles[players[0][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(398.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[0][1], self.x_percent(398.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(506), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+        self.profiles[players[1][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(711.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[1][1], self.x_percent(711.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(818.5), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+        self.profiles[players[2][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(1026.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[2][1], self.x_percent(1026.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(1133.5), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+
+    def createFourProfile(self, players):
```

Comment:
I'm not exactly sure if there is an easier way, because the locations of the profiles shift. I.e the location of the left most profile differs depending on the number of profiles and the spacing also differs. Moreover, the spacing isn't even consistent in the case there are 3 profiles (i.e the space between profile 1 and profile 2 is different from the space between profile 2 and profile 3). Otherwise, we could have a predefine ```x, y, spacing_offset``` depending on the number of players.

Reply:
Let's predefine the `x, y, spacing_offset` then